src/fight/mod.rs

```diff
@@ -3,157 +3,280 @@
 ///
-/// # Return
-/// usize: The number of attacks of the Regiment
-fn get_nb_attacks(regiment_attacking: &regiment::Regiment) -> usize {
-    (regiment_attacking.get_model().get_stats().get_attack() as f64
-        * regiment_attacking.get_cols() as f64
-        * 1.5)
-        .floor() as usize
+/// (&regiment::Regiment) defending_regiment: The defender
+///
+/// ## Return
+/// (usize, f64): A tuple with first the damage computed and then the probability that it occurs
+fn compute_mean_case(
+    attacking_regiment: &regiment::Regiment,
+    defending_regiment: &regiment::Regiment,
+) -> (usize, f64) {
+    let attacking_stats: &model::Stats = attacking_regiment.get_model().get_stats();
+    let defending_stats: &model::Stats = defending_regiment.get_model().get_stats();
+    let damage_probability: f64 =
+        computation_tools::compute_damage_probability(attacking_stats, defending_stats);
+    let nb_attacks =
+        (attacking_stats.get_attack() as f64 * 1.5 * attacking_regiment.get_cols() as f64).round();
+    let damage = std::cmp::min(
+        (nb_attacks * damage_probability).round() as usize,
+        defending_stats.get_health_point() * defending_regiment.get_nb_models(),
+    );
+
+    (
+        damage,
+        math_tools::compute_bernoulli(nb_attacks as usize, damage, damage_probability),
+    )
 }
 
-/// Compute the number of wounds a Regiment will suffer
+/// Find two thresholds for a gaussien assuring that the area covered reprenset at least 0.06% of the curve
 ///
-/// # Parameters
-/// nb_attacks (usize): The number of attacks of the attacking Regiment
+/// ## Parameters
+/// (usize) nb_touch: The number of hit that the attacker can assume
 ///
-/// to_hit (usize): The value to hit the defending Regiment
+/// (f64) wound_probability: The probability that a hit wound the enemy
 ///
-/// to_wound (usize): The value to wound the defending Regiment
+/// (usize) defender_hp: The amount of health point of the defender
 ///
-/// # Return
-/// usize: The number of wounds the defending Regiment will suffer
-fn get_nb_wounds(nb_attacks: usize, to_hit: usize, to_wound: usize) -> usize {
-    let nb_hit: usize = (((nb_attacks * (global_values::DEFAULT_DICE - to_hit + 1))
-        / global_values::DEFAULT_DICE) as f64)
-        .round() as usize;
-    (((nb_hit * (global_values::DEFAULT_DICE - to_wound + 1)) / global_values::DEFAULT_DICE) as f64)
-        .round() as usize
+/// ## Return
+/// (usize, usize): Our two gaussian threshold
+fn find_great_gauss_checkpoints(
+    nb_touch: usize,
+    wound_probability: f64,
+    defender_hp: usize,
+) -> (usize, usize) {
+    let max_hit = std::cmp::min(nb_touch, defender_hp);
+    let mut low_checkpoint: usize = (max_hit as f64 * (1.0_f64 / 3.0_f64)).round() as usize;
+    let mut high_checkpoint: usize = (max_hit as f64 * (2.0_f64 / 3.0_f64)).round() as usize;
+    let compute_proba = |e: usize| math_tools::compute_bernoulli(nb_touch, e, wound_probability);
```

Comment:
Can we static type the type of this variable ?

Reply:
I statically typed max_hit but not compute_proba. According to The Book:
A closure expression produces a closure value with a unique, anonymous type that cannot be written out.